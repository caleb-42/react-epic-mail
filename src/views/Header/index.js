import React from 'react';
import { connect } from "react-redux";
import { toggleNavigation } from '@redux/actions/navActions';
import { clearActiveMail } from '@redux/actions/mailActions';
import "hamburgers/_sass/hamburgers/hamburgers.scss";
import './index.scss';

const Header = ({
  className, activeNav, mails,
  toggleNavigation, clearActiveMail
}) => {
  const { active } = mails;
  const [navOpen, setNavOpen] = React.useState(false)
  const openNav = () => {
    toggleNavigation(!navOpen);
    setNavOpen(!navOpen)
  }
  return (
    <div className={`${className} ${(Object.keys(active).length > 0) ? 'post-active' : ''} flex relative z-20 header py-4 px-8 h-16`}>
      <a href="#" className="logo"></a>
      <div className="backbtn" onClick={clearActiveMail}></div>
      <h1 id="tabname" className="text-2xl text-white font-semibold self-center">{activeNav.subMenu || activeNav.menu}</h1>
      <button onClick={openNav} className={`hamburger hamburger--collapse ${navOpen ? 'is-active' : ''}`} type="button">
        <span className="hamburger-box">
          <span className="hamburger-inner"></span>
        </span>
      </button>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    activeNav: state.activeNav,
    mails: state.mails
  }
}

export default connect(mapStateToProps, { toggleNavigation, clearActiveMail })(Header);
