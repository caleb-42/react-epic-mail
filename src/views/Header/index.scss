.header{
  @apply flex justify-between items-center;
  background-image: url('/backgroundgrn.png');
  background-size: cover;
  background-repeat: no-repeat;
  #tabname{
    text-transform: capitalize;
  }
  .logo{
    width:4rem;
    height:2.5rem;
    @media  (min-width: 992px) {
      display: none;
    }
  }
  &.post-active .logo{
    @media  (max-width: 992px) {
      display: none;
    }
  }
  .backbtn{
    @apply hidden;
  }
  &.post-active .backbtn{
    @media  (max-width: 992px) {
      background-image: url('/backarrow-09.png');
      background-size: cover;
      background-repeat: no-repeat;
      @apply w-6 h-10 block;
    }
  }
  .hamburger{
    @apply hidden;
    @media  (max-width: 992px) {
      @apply p-1 text-white inline-block;
      .hamburger-inner, .hamburger-inner:after, .hamburger-inner:before{
        background: rgba(255,255,255,.7);
      } 
    }
  }
}