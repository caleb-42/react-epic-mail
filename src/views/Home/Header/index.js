import React from 'react';
import { connect } from "react-redux";
import './index.scss';

const Header = ({ className, activeNav }) => {
  return (
    <div className={`${className} flex header py-4 px-8 h-16`}>
      <div className="backbtn"></div>
      <h1 id="tabname" className="text-2xl text-white font-semibold self-center">{activeNav.subMenu || activeNav.menu}</h1>
      <div className="navicon img-con"></div>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    activeNav: state.activeNav
  }
}

export default connect(mapStateToProps, {})(Header);
