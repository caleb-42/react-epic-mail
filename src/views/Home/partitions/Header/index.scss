.header{
  background-image: url('/backgroundgrn.png');
  background-size: cover;
  background-repeat: no-repeat;
  #tabname{
    text-transform: capitalize;
  }
}