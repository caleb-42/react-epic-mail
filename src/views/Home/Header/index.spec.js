import React from "react";
import renderer from "react-test-renderer";
import { cleanup, render } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore from "@redux/configureStore";
import Header from "./index";

const store = configureStore();

const FakeHeader = (args) => {
  const defaultProps = {
    activeNav: { menu: 'Mails' }
  }
  const props = { ...defaultProps, ...args };
  const store = configureStore(props);
  return render(
    <ReduxProvider store={store}>
      <Header className="" />
    </ReduxProvider>
  )
}

describe('SNAPSHOT TEST', () => {

  it("Header Component", () => {
    const tree = renderer.create(
      <ReduxProvider store={store}>
        <Header className="" activeNav={{ menu: 'Mails' }} />
      </ReduxProvider >
    );
    expect(tree).toMatchSnapshot();
  });
})

describe('COMPONENT TEST', () => {

  afterEach(cleanup);

  it("should render menu name in Header", () => {
    const { getByText } = FakeHeader();
    getByText('Mails');
  });

  it("should render subMenu name in Header", () => {
    const { getByText } = FakeHeader({ activeNav: { menu: 'Mails', subMenu: 'Inbox' } });
    getByText('Inbox');
  });
})
