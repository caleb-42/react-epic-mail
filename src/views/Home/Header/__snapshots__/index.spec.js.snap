// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT TEST Header Component 1`] = `
<div
  className="  flex relative z-20 header py-4 px-8 h-16"
>
  <a
    className="logo"
    href="#"
  />
  <div
    className="backbtn"
    onClick={[Function]}
  />
  <h1
    className="text-2xl text-white font-semibold self-center"
    id="tabname"
  >
    Inbox
  </h1>
  <button
    className="hamburger hamburger--collapse "
    onClick={[Function]}
    type="button"
  >
    <span
      className="hamburger-box"
    >
      <span
        className="hamburger-inner"
      />
    </span>
  </button>
</div>
`;
